ui/src/app/stores/workflows/workflows.effects.ts

```diff
@@ -450,6 +452,81 @@ export class WorkflowsEffects {
     }),
   );
 
+  @Effect({ dispatch: true })
+  workflowExport = this.actions.pipe(
+    ofType(WorkflowActions.EXPORT_WORKFLOW),
+    switchMap((action: WorkflowActions.ExportWorkflow) => {
+      return this.workflowService.exportWorkflow(action.payload).pipe(
+        mergeMap((workflowBlobResponse: { blob: Blob; fileName: string }) => {
+          const a = document.createElement('a');
+          a.href = URL.createObjectURL(workflowBlobResponse.blob);
+
+          a.download = workflowBlobResponse.fileName.trim();
+          a.click();
+
+          this.toastrService.success(texts.EXPORT_WORKFLOW_SUCCESS_NOTIFICATION);
+          return [
+            {
+              type: EXPORT_WORKFLOW_DONE,
+            },
+          ];
+        }),
+        catchError(() => {
+          this.toastrService.error(texts.EXPORT_WORKFLOW_FAILURE_NOTIFICATION);
+          return [
+            {
+              type: EXPORT_WORKFLOW_DONE,
+            },
+          ];
+        }),
+      );
+    }),
+  );
+
+  @Effect({ dispatch: true })
+  workflowImport = this.actions.pipe(
+    ofType(WorkflowActions.IMPORT_WORKFLOW),
+    withLatestFrom(this.store.select(selectWorkflowState)),
+    switchMap(([action, state]: [WorkflowActions.ImportWorkflow, fromWorkflows.State]) => {
+      if (state.workflowAction.workflowFile) {
+        return this.workflowService.importWorkflow(state.workflowAction.workflowFile).pipe(
+          mergeMap((workflow: WorkflowJoinedModel) => {
+            this.toastrService.success(texts.IMPORT_WORKFLOW_SUCCESS_NOTIFICATION);
+            const workflowData = new WorkflowDataModel(workflow, state.workflowAction.workflowFormParts.dynamicParts);
+            return [
+              {
+                type: WorkflowActions.LOAD_WORKFLOW_SUCCESS,
```

Comment:
On the first occasion, importing a workflow takes me to create workflow page which is awesome and the workflow is persisted but on the second occasion (importing another workflow) I'm taken to a workflows/import page and the workflow is not persisted.

Reply:
Could you please try it one more time with all commits. I cannot reproduce it.